### PR TITLE
Add net.sourceforge.VMPK

### PR DESCRIPTION
**Pull request by hfiguiere**

### CODE REVIEW

**Comment by scx** [comment]
testbot, build net.sourceforge.VMPK

**Comment by scx** [comment]
testbot, build net.sourceforge.VMPK

**Comment by scx** [comment]
testbot, build net.sourceforge.VMPK

File: net.sourceforge.VMPK.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "http://downloads.sourceforge.net/project/vmpk/vmpk/0.7.1/vmpk-0.7.1.tar.bz2",
+          "sha256": "06d78872b8c23976cd1ed4e66ac6a3bfce93af28ce8db96c2c015cf185ce651b"
+        },
+        {
+          "type": "patch",
+          "path": "vmpk-hidpi.patch",
```
Comment (hfiguiere) [inline net.sourceforge.VMPK.json]: This patch has been submitted upstream https://sourceforge.net/p/vmpk/bugs/56/

**Comment by nedrichards** [comment]
bot, build net.sourceforge.VMPK
testbot, build net.sourceforge.VMPK

**Comment by nedrichards** [comment]
bot, build net.sourceforge.VMPK

**Comment by nedrichards** [comment]
@hfiguiere is the upstream interested in being involved? They seemed responsive to the first patch. Also, is there another one upstream to push the appdata?

**Comment by hfiguiere** [comment]
I haven't worked a patch yet to submit the appadata upstream. it's on my list

**Comment by nedrichards** [comment]
bot, build net.sourceforge.VMPK

(rebuilding to make sure the appdata passes the current suite before it goes upstream)

**Comment by bilelmoussaoui** [comment]
> @hfiguiere is the upstream interested in being involved? 

Any news regarding this? It seems to be the only blocker.
bot, build net.sourceforge.VMPK

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.VMPK

**Comment by hfiguiere** [comment]
The is the formal help request. https://sourceforge.net/p/vmpk/feature-requests/61/

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.VMPK

**Comment by pedrolcl** [comment]
I've tested the package, but it is not working:

```
pedro@linux-boccanegra:~> flatpak install --user https://dl.flathub.org/build-repo/8759/net.sourceforge.VMPK.flatpakref
Installing: net.sourceforge.VMPK/x86_64/test
Installing for user: net.sourceforge.VMPK/x86_64/test from net.sourceforge.VMPK-origin
[####################] 1 delta parts, 1 loose fetched; 861 KiB transferred in 1 seconds
pedro@linux-boccanegra:~> flatpak run net.sourceforge.VMPK 
qt.glx: qglx_findConfig: Failed to finding matching FBConfig (8 8 8 0)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig (1 8 8 0)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig (1 1 8 0)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig (1 1 1 0)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig (1 1 1 0)
No XVisualInfo for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
Falling back to using screens root_visual.
Failure loading Qt5 translations for "en_US" from "/usr/translations"
Failure loading VMPK translations for "en_US" from "/app/bin/../share/vmpk/locale/"
ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf
ALSA lib ../../../src/seq/seq.c:935:(snd_seq_open_noupdate) Unknown SEQ default
Error code: -2 ( No such file or directory )
Location: void drumstick::MidiClient::open(QString, int, bool)
terminate called after throwing an instance of 'drumstick::SequencerError'

```

Looks like the program is not able to access to the ALSA sequencer device node (/dev/snd/seq).

**Comment by hfiguiere** [comment]
I'm not sure what's happening. Here VMPK doesn't even try to access `/etc/alsa-pulse.conf`

Which version of flatpak do you have (`flatpak --version`) and which distro do you run on?

**Comment by pedrolcl** [comment]
> I'm not sure what's happening. Here VMPK doesn't even try to access `/etc/alsa-pulse.conf`
> 
> Which version of flatpak do you have (`flatpak --version`) and which distro do you run on?

opensuse leap 15.0
It comes with flatpak 0.10.4 but I've upgraded it to the latest available in community repositories

```
$ flatpak --version
Flatpak 1.0.5
```

Other paks I've tried:
* pulseeffects: works OK
* Hydrogen: no MIDI and no audio
```
$ flatpak run org.hydrogenmusic.Hydrogen
Qt: Session management error: Could not open network socket

Hydrogen 1.0.0-beta1 [Sep 30 2019]  [http://www.hydrogen-music.org]
Copyright 2002-2008 Alessandro Cominu
Copyright 2008-2018 The hydrogen development team
Hydrogen comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it under certain conditions. See the file COPYING for details.
ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf
ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf
ALSA lib ../../../src/seq/seq.c:935:(snd_seq_open_noupdate) Unknown SEQ hw
ALSA lib ../../../src/pcm/pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM hw:0
ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf
ALSA lib ../../../src/pcm/pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM default
(E) ::void* H2Core::alsaMidiDriver_thread(void*) Error opening ALSA sequencer: No such file or directory
(E) AlsaAudioDriver::connect ALSA: cannot open audio device hw:0:No such file or directory
(E) AlsaAudioDriver::connect ALSA: cannot open audio device default:No such file or directory
(E) ::void H2Core::audioEngine_startAudioDrivers() Error starting audio driver [audioDriver::connect()]
(E) ::void H2Core::audioEngine_startAudioDrivers() Using the NULL output audio driver
(E) ::void H2Core::audioEngine_startAudioDrivers() m_pMainBuffer_L == NULL
(E) ::void H2Core::audioEngine_startAudioDrivers() m_pMainBuffer_R == NULL
(E) NullDriver::setBpm not implemented yet
(E) Filesystem::check_permissions  is not a file
(E) Sample::load Unable to read                                                                                                                                                        
ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf                                                                                           
ALSA lib ../../../src/seq/seq.c:935:(snd_seq_open_noupdate) Unknown SEQ hw                                                                                                             
(E) ::void* H2Core::alsaMidiDriver_thread(void*) Error opening ALSA sequencer: No such file or directory                
```
* Muse3 same problem

```
$ flatpak run org.musesequencer.Muse3
Config File </home/pedro/.config/MusE/MusE.cfg>
configuration: unknown tag <dummyAudioBufSize> at line 21
configuration: unknown tag <dummyAudioSampleRate> at line 22
Qt: Session management error: Could not open network socket
LOCALE en_US
open projectfile: No such file or directory
Denormal protection enabled.
Select audio device from configuration : 0
User JackAudio backend - backend selected through configuration
MusE:initJackAudio: jack_get_version() returned zeros. Setting version major to 1.
cannot connect to jack server
cannot create jack client
Init RtAudioDevice
Error: RtAudio device selection illegal, setting up dummy audio backend!

RtAudio: no compiled support for specified API argument!

ALSA lib ../../src/conf.c:3956:(snd_config_update_r) Cannot access file /etc/alsa-pulse.conf
ALSA lib ../../../src/control/control.c:1375:(snd_ctl_open_noupdate) Invalid CTL hw:0

RtApiAlsa::getDeviceCount: control open, card = 0, No such file or directory.
```

**Comment by hfiguiere** [comment]
Apparently there is some info there:
https://github.com/flatpak/flatpak/issues/1875

And how to solve it.

**Comment by barthalion** [comment]
Bumping Bilel's question – could you drop upstream a message? Given you're a proven maintainer we don't need to wait for their answer.

**Comment by hfiguiere** [comment]
See https://github.com/flathub/flathub/pull/855#issuecomment-538665361
also pedrolcl above is upstream.

**Comment by barthalion** [comment]
Ah, my bad! Thanks.

/merge @pedrolcl

**Comment by barthalion** [comment]
FYI @pedrolcl, you will receive an invitation to collaborators. Make sure to accept it to get write access to the repository above.

**Comment by hfiguiere** [comment]
Thanks !

**Pull request closed without merging**